visualization/app/codeCharta/util/fileValidator.ts

```diff
@@ -35,41 +35,57 @@ export const ERROR_MESSAGES = {
 export function validate(nameDataPair: NameDataPair) {
 	const file = nameDataPair?.content
 	const result: CCValidationResult = { error: [], warning: [] }
+
+	result.error = checkErrors(file)
+	result.warning = checkWarnings(file)
+
+	if (result.error.length > 0) {
+		throw result.error
+	}
+}
```

Comment:
I propably would call `checkErrors` and `checkWarnings` in loadFiles directly. The `validate` function is not needed anymore. And instead of throwing errors and warnings from one function to another, I would just process the return values of the two new check functions.

Reply:
Awesome that would simplify the code even further!